kernel/yosys.cc

```diff
@@ -540,6 +540,25 @@ std::string escape_filename_spaces(const std::string& filename)
 	return out;
 }
 
+#ifdef _WIN32
+const char* const OS_PATH_SEP = "/\\";
+#else
+const char* const OS_PATH_SEP = "/";
+#endif
+
+std::string name_from_file_path(std::string path) {
+	size_t sep_pos = path.find_last_of(OS_PATH_SEP);
+	if (sep_pos != std::string::npos)
+		return path.substr(sep_pos + 1);
+	else
+		return path;
+}
+
+// Includes OS_PATH_SEP at the end
+std::string parent_from_file_path(std::string path) {
+	return path.substr(0, path.find_last_of(OS_PATH_SEP)+1);
```

Comment:
What happens if `path.find_last_of(OS_PATH_SEP)` returns `std::string::npos`?

Reply:
To me it looks like it'll return `""`, which does not include `OS_PATH_SEP` at the end. This either needs a documentation change or an assertion.